straxen/plugins/bayes_peaklet_classification.py

```diff
@@ -0,0 +1,138 @@
+import numpy as np
+import numba
+from scipy.special import logsumexp
+import strax
+import straxen
+
+
+class BayesPeakletClassification(strax.Plugin):
+    """
+    Bayes Peaklet classification
+    Returns the ln probability of a each event belonging to the S1 and S2 class.
+    Uses conditional probabilities and data parameterization learned from wfsim data.
+
+    :param peaks: peaks
+    :param waveforms: peaks waveforms in PE/ns
+    :param quantiles: quantiles in ns
+    :returns: the ln probability of a each peaklet belonging to S1 and S2 class
+    """
+    provides = 'peak_classification_bayes'
+    depends_on = ('peaks',)
+    __version__ = '0.0.1'
+    dtype = (strax.time_fields
+             + [('ln_prob_s1', np.float32, 'S1 ln probability')]
+             + [('ln_prob_s2', np.float32, 'S2 ln probability')]
+            )
+
+    # Descriptor configs
+    bayes_config_file = straxen.URLConfig(
+        default='resource://cmt://'
+                'bayes_model'
+                '?version=ONLINE&run_id=plugin.run_id&fmt=npy',
+        help='Bayes model, conditional probabilities tables and Bayes discrete bins'
+    )
+    bayes_n_nodes = straxen.URLConfig(
+        default=50,
+        help='Number of attributes(features) per waveform and quantile'
+    )
+    n_bayes_classes = straxen.URLConfig(
+        default=2,
+        help='Number of label classes S1(1)/S2(2)'
+    )
+
+    def setup(self):
+
+        self.class_prior = np.ones(self.n_bayes_classes)/self.n_bayes_classes
+        self.bins = self.bayes_config_file['bins']
+        self.cpt = self.bayes_config_file['cprob']
+
+    def compute(self, peaks):
+
+        waveforms, quantiles = compute_wf_and_quantiles(peaks, self.bayes_n_nodes)
+
+        ln_prob_s1, ln_prob_s2 = compute_inference(self.bins, self.bayes_n_nodes, self.cpt,
+                                                   self.n_bayes_classes, self.class_prior,
+                                                   waveforms, quantiles)
+
+        return dict(time=peaks['time'],
+                    endtime=peaks['time'] + peaks['dt'] * peaks['length'],
+                    ln_prob_s1=ln_prob_s1,
+                    ln_prob_s2=ln_prob_s2
+                    )
+
+def compute_wf_and_quantiles(peaks: np.ndarray, bayes_n_nodes: int):
```

Comment:
I think this function can be numbafied easily. Within your for-loop your can use a fixed length buffer instead of np.append as the length of your buffer is determined by the number of quartiles  and peaks is not it? The other thing which your have to change is `np.linspace(0., 1., bayes_n_nodes, endpoint=False)` into `np.linspace(0., 1., bayes_n_nodes)[:-1]` (assuming this is what endpoint false does).

Reply:
I did an attempt to get rid of the np.append following your suggestion 
```
# create buffer to avoid np.append
frac_of_cumsum = np.empty(num_samples+1)
frac_of_cumsum[0] = 0.0
frac_of_cumsum[1:] = np.cumsum(data[i, :]) / np.sum(data[i, :])
# create buffer to avoid np.append
cumsum_steps = np.empty(bayes_n_nodes+1)
cumsum_steps[0:bayes_n_nodes] = np.interp(np.linspace(0., 1., bayes_n_nodes, endpoint=False), frac_of_cumsum, sample_number)
 cumsum_steps[-1:] = sample_number[-1]
quantiles[i, :] = cumsum_steps[1:] - cumsum_steps[:-1]
```

Locally passed all tests, however did not pass test on GitHub

Reply:
You mean with njit decorator? I think it cannot be numbafied due to `, endpoint=False` in linespace. You can check numbas very excellent [reference manual](https://numba.readthedocs.io/en/stable/reference/numpysupported.html).

Reply:
My first attempt was to remove the np.append(), once that is working I can try to numbafied